<a href="https://colab.research.google.com/github/michaelherediaperez/mecanica_de_solidos_un/blob/main/codigos/cap_04/04_08_03_ejemplo_DP.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 4.8.3. Relación entre los esfuerzos principales obtenidos en el análisis bidimensional y tridimensional 

|Quién | Fecha | Qué hizo |
| ---  | ---   | ---      |
|Michael Heredia Pérez, <mherediap@unal.edu.co> | Julio 20, 2022 | Conversión de código de MATLAB a Python |
|Michael Heredia Pérez, <mherediap@unal.edu.co> | Octubre 12, 2022 | Completando el código |

## Ejemplo de *deformación plana*


In [3]:
import numpy as np
#np.set_printoptions(precision=4)   # imprime todo con 4 cifras decimales max

Asumamos $\nu = 0.30$ y un estado de **deformación plana**; para calcular los nuevos esfuerzos y direcciones principales, adicionalmente a $\sigma_x = $ − 1 Pa, $\sigma_y = $ = 2 Pa y $\tau_{xy} = $ − 3 Pa, se debe tener en cuenta que $\tau_{xz} = G\gamma_{xz}$ = 0 Pa, $\tau_{yz} = G\gamma_{yz}$ = 0 Pa y $\sigma_z = \nu(\sigma_x + \sigma_v) = $ 0.3( − 1 Pa + 2 Pa) = 0.3 Pa (este último esfuerzo se calculó utilizando la ecuación (4.42)). Para calcular los valores y vectores propios del problema ensamblamos la matriz (4.43)

In [9]:
# Defino las variables.
sx  = -1
sy  =  2
txy = -3
nu  = 0.30

# La matriz de esfuerzos.
sigma = np.array([[sx, txy,            0],
                  [txy, sy,            0],
                  [0,    0, nu*(sx + sy)]])

In [10]:
sigma

array([[-1. , -3. ,  0. ],
       [-3. ,  2. ,  0. ],
       [ 0. ,  0. ,  0.3]])

In [11]:
# Calculo lo vectores y valores propios
valp, vecp = np.linalg.eigh(sigma)

In [12]:
# Los valores propios salen ordenados de menor a mayor [s3, s2, s1]
valp

array([-2.8541,  0.3   ,  3.8541])

In [13]:
# Los vectores propios correspondientes [n3, n2, n1]
vecp

array([[-0.8507,  0.    , -0.5257],
       [-0.5257,  0.    ,  0.8507],
       [-0.    ,  1.    ,  0.    ]])

Verificamos que los vectores principales formen un sistema coordenado de la mano derecha, a partir del producto vectorial ($\mathbf{ \hat{n}_3 = \hat{n}_1 \times \hat{n}_2 }$):

In [14]:
# Extraigo las direcciones principales del cálculo anterior.
n3, n2, n1 = vecp.T

# Calculo n3 = n1 x n2
n3_calculado = np.cross(n1, n2)
n3_calculado

array([ 0.8507,  0.5257, -0.    ])

Fíjese que obtenemos el vector $\mathbf{\hat{n}}_3$ con sentido contrario, esto se puede solucionar considerando en sentido contrario a $\mathbf{\hat{n}}_2$

In [15]:
# Calculo n3 = n1 x -n2
n3_calculado = np.cross(n1, -n2)
n3_calculado

array([-0.8507, -0.5257,  0.    ])

A diferencia del código de Matlab visto en el libro guía, aquí el ajuste de signos es diferente.

## Conclusión de main.pdf

Observe que en este caso, dos de los valores propios coincidieron con $(\sigma_1)_{xy} =$ 3.8541 Pa y $(\sigma_2)_{xy} =$ − 2.8541 Pa, mientras que el otro valor propio es igual al $\sigma_z$ de
la ecuación (4.42), es decir, $\sigma_ = \nu (\sigma_x + \sigma_y)$ . De hecho, las direcciones principales asociadas a tales valores propios describieron los vectores $[ − 0.5257, 0.8507, 0]^\intercal$ y
$[0.8507, 0.5257, 0]^\intercal$ que habíamos mencionado anteriormente y el vector propio asociado al valor propio de $\sigma_z$ fue $[0, 0, − 1]^\intercal$ (ver Figura 4.17).

Finalmente, debemos recalcular el esfuerzo cortante máximo. Según se dijo en la Sección 2.9.7, este tiene una magnitud dada por la ecuación (2.75): $\tau_{max} = \frac{\sigma_1-\sigma_3}{2}$ y se presenta en los planos ortogonales a los vectores $\frac{ \hat{n}_1 - \hat{n}_3}{||\hat{n}_1 - \hat{n}_3||}$ y $\frac{ \hat{n}_1 + \hat{n}_3}{||\hat{n}_1 + \hat{n}_3||}$; por lo tanto, el esfuerzo cortante máximo es 3.3541 Pa y se presenta en los planos con normal $\frac{ \hat{n}_1 - \hat{n}_3}{||\hat{n}_1 - \hat{n}_3||} = $ $[0.2298, 0.9732, 0]^\intercal$ y
$\frac{ \hat{n}_1 + \hat{n}_3}{||\hat{n}_1 + \hat{n}_3||} =$  $[ − 0.9732, 0.2298, 0]^\intercal$.

En resumen, se tiene que dos de los valores propios corresponden a $(\sigma_1)_{xy}$ y $(\sigma_2)_{xy}$, mientras que el otro corresponde a $\sigma_z$. Los vectores propios asociados son los mismos calculados en el caso bidimensional para los dos primeros, agregando un 0 en la componente $z$, mientras que el vector propio asociado a $\sigma_z$ tiene la dirección de $[0,0, \pm 1]^\intercal$ , siendo necesario ajustar su sentido. 

De este modo, luego de reordenar los valores propios, tenemos:

$$
\begin{align*}

\sigma_1    &= \max\left( (\sigma_1)_{xy}, (\sigma_1)_{xy}, \nu(\sigma_x + \sigma_y) \right)  \\
\sigma_2    &= mediana\left( (\sigma_1)_{xy}, (\sigma_1)_{xy}, \nu(\sigma_x + \sigma_y) \right) \\
\sigma_3    &= \min\left( (\sigma_1)_{xy}, (\sigma_1)_{xy}, \nu(\sigma_x + \sigma_y) \right) \\
\tau_{\max} &= \max\left( \frac{(\sigma_1)_{xy} - \nu(\sigma_x + \sigma_y)}{2},  \frac{(\sigma_2)_{xy} - \nu(\sigma_x + \sigma_y)}{2},  \frac{ (\sigma_1)_{xy} - (\sigma_2)_{xy} }{2}   \right)  

\end{align*}
$$
